sys/include/xtimer.h

```diff
@@ -479,7 +479,16 @@ static inline void xtimer_spin_until(uint32_t target) {
 
 static inline void xtimer_spin(uint32_t offset) {
     uint32_t start = _lltimer_now();
-    while ((_lltimer_now() - start) < offset);
+    uint32_t target = start + offset;
+#if XTIMER_MASK
+    target = _lltimer_mask(target);
+#endif
+    if (target < start) {
+        /* Target time is less than start time => overflow */
+        /* Spin until timer overflows */
+        while (_lltimer_now() >= start);
```

Comment:
no, because then if the low level timer has not yet incremented since the start tick then we will immediately jump out of the spin.